index/secondary_index_manager.cc

```diff
@@ -100,6 +100,10 @@ sstring index_table_name(const sstring& index_name) {
     return format("{}_index", index_name);
 }
 
+sstring index_name_from_table_name(const sstring& table_name) {
+    return table_name.substr(0, table_name.size() - 6); // remove the _index suffix from an index name;
```

Comment:
It would have been better to check and throw an exception if the table_name doesn't end with "_index". It's a very cheap test (this is a very rarely called function) and can help serious bugs in using this function.